source/controllers/README.md

```diff
@@ -1,122 +1,246 @@
+
+```c++
+#include "controllers/ControllerFactory.hpp"
+#include "state_representation/parameters/Parameter.hpp"
 
-```cpp
+using namespace controllers;
 using namespace state_representation;
 
-Dissipative<CartesianState> task_space_dissipative;
-CartesianTwist desired_twist;
-CartesianTwist real_eef_twist;
-Jacobian robot_jacobian;
+std::list<std::shared_ptr<ParameterInterface>> parameters;
+parameters.emplace_back(make_shared_parameter("damping", 10.0));
+parameters.emplace_back(make_shared_parameter("stiffness", 5.0));
+parameters.emplace_back(make_shared_parameter("inertia", 1.0));
+
+auto ctrl = CartesianControllerFactory::create_controller(CONTROLLER_TYPE::IMPEDANCE, parameters);
+```
+
+## Using a controller
+
+The controller factory returns a base class pointer `shared_ptr<IController<S>>` for the chosen state type `S`,
+pointing to an instance of a specific derived controller. This allows all controllers to share the same consistent
+interface for usage and configuration.
+
+### Parameters
+
+Controllers support custom parameterization through the base interface using `state_representation::ParameterMap`
+methods:
+- `get_parameters()`
+- `get_parameter(name)`
+- `get_parameter_value<T>(name)`
+- `set_parameters(parameters)`
+- `set_parameter(parameter)`
+- `set_parameter_value(parameter, value)`
+
+These methods can be used after construction to get or set controller parameters. Refer to the documentation
+on `controllers::IController<>` and `state_representation::ParameterMap` for more information.
+
+For controllers that use a `robot_model::Model` object, there are additional interface methods that can be used
+to get or set the current robot model. Note that changing the robot model dimensionality, reference frame or joint
+frames at runtime could cause unexpected behaviour.
+- `get_robot_model()`
+- `set_robot_model(robot_model)`
+
+### Compute command
+
+The main purpose of each controller is the `compute_command` function, which calculates an output state from a 
+desired command state and feedback state. For example, the impedance controller takes the error between the command
+state and the feedback state and multiplies it by some linear gains to compute the output state. 
 
-JointTorques command = task_space_dissipative.compute_command(desired_twist,
-                                                              real_eef_twist,
-                                                              jacobian)
+```c++
+#include "controllers/ControllerFactory.hpp"
+#include "state_representation/space/cartesian/CartesianState.hpp"
+
+using namespace controllers;
+
+// create a Cartesian impedance controller
+auto ctrl = CartesianControllerFactory::create_controller(CONTROLLER_TYPE::IMPEDANCE);
+
+auto command_state = CartesianState::Random("command");
+auto feedback_state = CartesianState::Random("feedback");
+
+// compute the command output
+auto command_output = ctrl->compute_command(command_state, feedback_state);
```

Comment:
The `auto` is nice and it's how we would use it but maybe for a README, we should be more explicity about what the function returns? Line 119 and 120 are okay because you can see its a CartesianState but here it would maybe be nicer to put CartesianState explicitely and in the next example JointState for the result of compute_command. But thats not critical, if you prefer `auto` thats fine